cmds/core/timeout/main.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2012-2022 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Run a command and kill it if it runs more than a specified duration
+//
+// Synopsis:
+//	timeout [-t duration-string] command [args...]
+//
+// Description:
+//	timeout will run the command until it succeeds or too much time has passed.
+//	The default timeout is 30s.
+//	If no args are given, it will print a usage error.
+//
+// Example:
+//	$ timeout echo hi
+//	hi
+//	$
+//	$./timeout -t 5s bash -c 'sleep 40'
+//	$ 2022/03/31 14:47:32 signal: killed
+//	$./timeout  -t 5s bash -c 'sleep 40'
+//	$ 2022/03/31 14:47:40 signal: killed
+//	$./timeout  -t 5s bash -c 'sleep 1'
+//	$
+
+//go:build !test
+// +build !test
+
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"log"
+	"os"
+	"os/exec"
+	"time"
+)
+
+type cmd struct {
+	args         []string
+	timeout      time.Duration
+	in, out, err *os.File
+}
+
+var (
+	timeout        = flag.Duration("t", 30*time.Second, "Timeout for command")
+	errNoArgs      = errors.New("Need at least a command to run")
+	errBadDuration = errors.New("timeout must be a duration")
+)
+
+func main() {
+	flag.Parse()
+	c := &cmd{args: flag.Args(), in: os.Stdin, out: os.Stdout, err: os.Stderr, timeout: *timeout}
+	if err := c.run(); err != nil {
+		log.Fatalf("timeout(%v):%v", *timeout, err)
+	}
+}
+
+func (c *cmd) run() error {
+	if len(c.args) == 0 {
+		return errNoArgs
+	}
+	ctx := context.Background()
+	ctx, cancel := context.WithDeadline(ctx, time.Now().Add(c.timeout))
```

Comment:
any reason not to use context.WithTimeout() - that way you can avoid the `time.Now().Add()` stuff.

Reply:
Thanks, good catch. Done.